### PR TITLE
Fix sync batch method coverage

### PR DESCRIPTION
**Pull request by christothes**
Instrument the `TableTransactionalBatch` in tests and make supporting methods virtual so that they are properly proxied.

### CODE REVIEW

File: sdk/tables/Azure.Data.Tables/src/TableTransactionalBatch.cs [hunk 1]
```diff
@@ -300,7 +300,7 @@ private bool AddMessage(ITableEntity entity, HttpMessage message, RequestType re
         /// Builds an ordered list of <see cref="HttpMessage"/>s containing the batch sub-requests.
         /// </summary>
         /// <returns></returns>
-        private List<(ITableEntity entity, HttpMessage HttpMessage)> BuildOrderedBatchRequests()
+        internal virtual List<(ITableEntity entity, HttpMessage HttpMessage)> BuildOrderedBatchRequests()
```
Comment (pakrym) [inline sdk/tables/Azure.Data.Tables/src/TableTransactionalBatch.cs:303]: Are you using this method in test? Why does it need to be proxied?

Comment (christothes) [reply sdk/tables/Azure.Data.Tables/src/TableTransactionalBatch.cs:303]: the result of the method is used [here](https://github.com/Azure/azure-sdk-for-net/blob/01793bf2b016e410bda508034128e7172cdc07a8/sdk/tables/Azure.Data.Tables/src/TableTransactionalBatch.cs#L222). Before this was proxied `_submittedMessageList` would be null.

Comment (christothes) [reply sdk/tables/Azure.Data.Tables/src/TableTransactionalBatch.cs:303]: And it is referenced by the test [here](https://github.com/Azure/azure-sdk-for-net/blob/01793bf2b016e410bda508034128e7172cdc07a8/sdk/tables/Azure.Data.Tables/tests/TableClientLiveTests.cs#L1100)

Comment (pakrym) [reply sdk/tables/Azure.Data.Tables/src/TableTransactionalBatch.cs:303]: The call at https://github.com/Azure/azure-sdk-for-net/blob/01793bf2b016e410bda508034128e7172cdc07a8/sdk/tables/Azure.Data.Tables/tests/TableClientLiveTests.cs#L1100 should be itself proxied to the base

Comment (christothes) [reply sdk/tables/Azure.Data.Tables/src/TableTransactionalBatch.cs]: You are correct, I only needed `TryGetFailedEntityFromException to be virtual.

**Review by pakrym — APPROVED** [review]

**Pull request merged**